featurebyte/service/observation_table.py

```diff
@@ -158,50 +287,30 @@ async def validate_materialized_table_and_get_metadata(
         Returns
         -------
         dict[str, Any]
-
-        Raises
-        ------
-        MissingPointInTimeColumnError
-            If the point in time column is missing.
-        UnsupportedPointInTimeColumnTypeError
-            If the point in time column is not of timestamp type.
-        ValueError
-            If no entity column is provided.
         """
+        # Get column info and number of row metadata
         columns_info, num_rows = await self.get_columns_info_and_num_rows(
             db_session=db_session,
             table_details=table_details,
             serving_names_remapping=serving_names_remapping,
         )
-        columns_info_mapping = {info.name: info for info in columns_info}
-
-        if SpecialColumnName.POINT_IN_TIME not in columns_info_mapping:
-            raise MissingPointInTimeColumnError(
-                f"Point in time column not provided: {SpecialColumnName.POINT_IN_TIME}"
-            )
-
-        point_in_time_dtype = columns_info_mapping[SpecialColumnName.POINT_IN_TIME].dtype
-        if point_in_time_dtype not in {
-            DBVarType.TIMESTAMP,
-            DBVarType.TIMESTAMP_TZ,
-        }:
-            raise UnsupportedPointInTimeColumnTypeError(
-                f"Point in time column should have timestamp type; got {point_in_time_dtype}"
-            )
-
-        if not skip_entity_validation_checks:
-            # Check that there's at least one entity mapped in the columns info
-            if not any(info.entity_id is not None for info in columns_info):
-                raise ValueError("At least one entity column should be provided.")
-
-        most_recent_point_in_time = await ObservationTableService.get_most_recent_point_in_time(
+        # Perform validation on column info
+        validate_columns_info(columns_info, skip_entity_validation_checks)
+        # Get point in time metadata
+        point_in_time_stats = await get_point_in_time_stats(
             db_session=db_session,
             destination=table_details,
         )
+        # Get entity statistics metadata
+        column_name_to_count = await self._get_column_name_to_entity_count(
+            feature_store, table_details, columns_info
+        )
         return {
             "columns_info": columns_info,
             "num_rows": num_rows,
-            "most_recent_point_in_time": most_recent_point_in_time,
+            "most_recent_point_in_time": point_in_time_stats.most_recent,
+            "earliest_point_in_time": point_in_time_stats.earliest,
```

Comment:
nit: I feel `least_recent_point_in_time` is a more natural counterpart to `most_recent_point_in_time`, but feel free to keep it this way if you don't think so.